plugins/shadow-block-converter/src/shadow_block_converter.ts

```diff
@@ -72,7 +75,7 @@ export class BlockShadowStateChange extends Blockly.Events.BlockBase {
    */
   toJson(): BlockShadowStateChangeJson {
     const json = super.toJson() as BlockShadowStateChangeJson;
-    if (!this.connectionIndex) {
+    if (typeof this.inputIndexInParent === 'undefined') {
```

Comment:
```suggestion
    if (this.inputIndexInParent === undefined) {
```

(Unless you are worried that someone might try to assign a different value to the `undefined` global variable, but this has been disallowed since at least ES5.1!)

Reply:
This check isn't needed now that the properties and constructor arguments can't be undefined.